core/lib/zksync_core/src/api_server/web3/backend_jsonrpsee/metadata.rs

```diff
@@ -0,0 +1,145 @@
+//! Method metadata.
+
+use std::{cell::RefCell, mem, sync::Arc, time::Instant};
+
+use thread_local::ThreadLocal;
+use zksync_types::api;
+use zksync_web3_decl::{error::Web3Error, jsonrpsee::MethodResponse};
+
+#[cfg(test)]
+use super::testonly::RecordedMethodCalls;
+use crate::api_server::web3::metrics::API_METRICS;
+
+/// Metadata assigned to a JSON-RPC method call.
+#[derive(Debug, Clone)]
+pub(crate) struct MethodMetadata {
```

Comment:
`MethodMetadata` is essentially a request-scoped variable (using terminology from Spring). It's propagated to RPC call handlers using thread-local variables (using [`ThreadLocal`](https://docs.rs/thread_local/latest/thread_local/struct.ThreadLocal.html) rather than `thread_local!` from the standard library in order to be less "ambient").